### PR TITLE
Fix parsing copy constructor

### PR DESCRIPTION
**Pull request by Adda0**
This PR optimizes parsing by introducing move assignment operator and move constructor for `FormulaGraph` and requiring some moves to happen during the parsing. It is just a quick improvement for now. Modifications to the overall algorithms are needed which will probably completely rewrite these functions.

@martinhruska Feel free to give me your input on these changes. As discussed privately, this helps, but is not enough.

### CODE REVIEW

**Comment by Adda0** [comment]
I will have a look what the MacOS error is tomorrow.

File: src/inter-aut.cc [hunk 1]
```diff
@@ -308,7 +309,7 @@ namespace {
      * @param postfix Postfix to which operators are eventually added
      * @return A postfix with eventually added operators
      */
-    std::vector<mata::FormulaNode> add_disjunction_implicitly(const std::vector<mata::FormulaNode> &postfix)
+    std::vector<mata::FormulaNode> add_disjunction_implicitly(std::vector<mata::FormulaNode> postfix)
```
Comment (tfiedor) [inline src/inter-aut.cc:310]: Why did you remove the `&` reference? Isn't this less efficient?

**Review by tfiedor — APPROVED** [review]
Seems ok

File: src/inter-aut.cc (hunk 1 above)
Comment (Adda0) [reply src/inter-aut.cc:310]: Because we are moving the data from the caller to the function parameter. Hence, we can afterwards move the elements of the postfix vector into the output vector. Otherwise, we would have to create a new copy in the output vector.

**Comment by Adda0** [comment]
Due to some new surprising discoveries, I ask the reviews to wait with their reviews. Nothing much should change, but there might be some new modifications introduced.

**Comment by Adda0** [comment]
The changes since the last time have great impact on the performance, but are minuscule in the code itself. The parser should be about 10.5 times faster on average than the current parser. On the examples I tried, we get from 3.8 s to 0.36 s. It might still be too slow for our needs, but it is definitely better. At least, it is a first step.   

@jurajsic Could you please look at the PR and review the changes now?

**Review by jurajsic — APPROVED** [review]
